### PR TITLE
[HWORKS-740] Expat - disable provenance on existing projects

### PR DESCRIPTION
**Pull request by o-alex**

### CODE REVIEW

File: src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+      setXAttr(path, new ProvCoreDTO(Provenance.Type.FULL.dto, null));
+    };
+  }
+  
+  private String datasetPath(String projectName, String datasetName) {
+    if(datasetName.equals(projectName.toLowerCase() + ".db") ||
+      datasetName.equals(projectName.toLowerCase() + "_featurestore.db")) {
+      return "/apps/hive/warehouse/" + datasetName;
+    } else {
+      return "/Projects/" + projectName + "/" + datasetName;
```
Comment (bubriks) [inline src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java:236]: could use `projectPath(String projectName)`

**Review by bubriks — CHANGES_REQUESTED** [review]

File: src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java (hunk 1 above)
Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java:236]: fixed

File: src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+  }
+  
+  private CheckedConsumer<ProcessState, Exception> rollbackDataset() {
+    return state -> {
+      Path path = new Path(datasetPath(state.projectName, state.datasetName));
+      if (state.datasetName.equals(state.projectName + "_Training_Datasets") ||
+        state.datasetName.equals(state.projectName.toLowerCase() + "_featurestore.db")) {
+        dfso.setMetaStatus(path, Inode.MetaStatus.META_ENABLED);
+        setXAttr(path, new ProvCoreDTO(Provenance.Type.META.dto, state.projectInodeId));
+      } else if (state.datasetName.equals(state.projectName.toLowerCase() + ".db")) {
```
[the remark this answers is not in the document]
Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java]: true, but I prefer to keep it explicit so I see that I dealt with all dataset types.

File: src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        dfso.setMetaStatus(path, Inode.MetaStatus.FULL_PROV_ENABLED);
+        setXAttr(path, new ProvCoreDTO(Provenance.Type.FULL.dto, state.projectInodeId));
+      } else {
```
[the remark this answers is not in the document]
Comment (o-alex) [reply src/main/java/io/hops/hopsworks/expat/migrations/featurestore/metadata/DisableEpipeMigration.java]: switch cases require constant expressions such as `Models`. You cannot build strings in a case such as `projectName+".db"`

**Review by bubriks — APPROVED** [review]
:+1:

**Pull request merged**